### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by aandriyenko-lohika**
# Lesson 2

### Homework

1. Migrate to typescript
2. Create server.ts with couple of routes
  * Use different http methods
  * Implement extract of the request body
3. Add error handlers (can be done in express server)
  * Unhandled rejection
  * Uncaught exception
4. Add logger which would also log request details (can be done   in express server)
 * Update logger by using vendor lib instead of console
 * Log request details (request id, url, params)
 * Implement at least one transport (can be to the file) *

### Run project
1. `npm run start:dev`

### CODE REVIEW

File: lesson2/src/server.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import { PORT } from './config';
+import logger from './logger';
+
+function getRawBody(req: http.IncomingMessage): Promise<string> {
+  const chunks: Uint8Array[] = [];
+  return new Promise((resolve, reject) => {
+    req
+      .on('data', (chunk) => {
+        chunks.push(chunk);
```
Comment (franxx32) [inline lesson2/src/server.ts:11]: You can use async iterators without subscribing on different event
```js
for await (const chunk of req) {
      body.push(chunk);
 }
```

File: lesson2/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import logger from './logger';
+
+export const PORT = process.env.PORT || 3000;
+
+logger.info(process.env)
+
+process.on('uncaughtException', (error) => {
```
Comment (franxx32) [inline lesson2/src/config.ts:7]: config.ts is not so suitable for error handling

**Review by franxx32 — APPROVED** [review]

File: lesson2/src/server.ts (hunk 1 above)
Comment (aandriyenko-lohika) [reply lesson2/src/server.ts]: Thanks for the advice. It looks much cleaner now!

File: lesson2/src/config.ts (hunk 2 above)
Comment (aandriyenko-lohika) [reply lesson2/src/config.ts]: Thanks for the comment. I moved both error handlers to the `errorHandlers.ts` file.

**Pull request merged**